docs/core/extensions/snippets/configuration/console-json/ValidateSettingsOptions.cs

```diff
@@ -14,26 +15,26 @@ class ValidateSettingsOptions : IValidateOptions<SettingsOptions>
 
     public ValidateOptionsResult Validate(string name, SettingsOptions options)
     {
-        string result = "";
+        StringBuilder failure = new();
         var rx = new Regex(@"^[a-zA-Z''-'\s]{1,40}$");
```

Comment:
Unrelated to the changes, but I think it makes more sense to make this a private static field so that allocation is done only once. The private static field could be directly initialized or could be made lazy. I don't have a strong opinion which is better.

Reply:
I do not like the idea of having a class-scoped variable as we do not control the number of times that `Validate` is called, nor do we know from what number of threads this is called concurrently. Not saying it's an issue, but it leaves the door open for the contention and stale reads/writes.